### PR TITLE
Add Expected Start Semester to Educator Profile

### PR DESCRIPTION
**Pull request by mwvolo**
## Summary

Adds an optional "When do you plan to start using OpenStax?" dropdown to the educator profile completion form. Value is captured on `users.expected_start_semester`, emitted as a property on the `educator_complete_profile` PostHog event, and synced to the Salesforce `Lead` object. Conditional visibility: shown only for `as_primary` / `as_recommending` paths; hidden for `as_future`. Gated by `Settings::FeatureFlags.expected_start_semester_enabled` (default `false`).

Jira: [DATA-301](https://openstax.atlassian.net/browse/DATA-301) (epic: DATA-304)

## 🚧 Blocked on Salesforce admin work

**Do not merge until all four items below are complete in production Salesforce.** Rails code assumes these exist; without them, `lead.expected_start_semester = ...` will raise at runtime.

- [x] ` Expected_Start_Semester__c` picklist created on **Lead** object
- [x] Same field created on **Contact** object
- [x] Lead → Contact field mapping configured for that field
- [x] Picklist values match exactly: ` This semester`, ` Next semester`, ` Next academic year`, ` Just exploring`

## Architecture

One-form feature layered onto existing surfaces — no new routes, models, or controllers. The view fieldset is wrapped in the feature flag; the handler adds a triple-guard helper (blank / allow-list / path match) so the value is only persisted for adopter paths and only valid strings land in the DB. The SF routine translates the snake_case storage value into the human picklist label via a constant map (`EXPECTED_START_SEMESTER_LABELS`) next to the existing ` ADOPTION_STATUS_FROM_USER` sibling.

## Deploy sequence

1. Merge & deploy with flag **` false`** (default).
2. Verify prod migration ran cleanly.
3. Flip flag via Rails console: ` Settings::FeatureFlags.expected_start_semester_enabled = true`.
4. Smoke-test one real profile completion end-to-end (form → PostHog → SF Lead).

## Rollback

` Settings::FeatureFlags.expected_start_semester_enabled = false` in Rails console — takes effect immediately. Already-captured values stay on ` users`, in PostHog, and in SF. Rollback hides the field, it does not erase data.

## Test plan

- [x] Handler spec: 7 new examples covering all three ` using_openstax_how` paths + edge cases (blank, unknown value, omitted field, path-guard rejection).
- [x] Controller spec: 2 new examples asserting PostHog emission (with and without a selected value).
- [x] Controller spec: 3 new examples asserting view rendering (flag on/off + ` display: none` initial state).
- [x] SF routine spec: 6 new examples covering all four label mappings + nil + unrecognized value.
- [x] Feature spec (` js: true`): 4 new examples for visibility across ` as_primary` / ` as_recommending` / ` as_future` + clear-on-switch + flag-off.
- [x] Full targeted suite (handler + controller + SF routine + feature): 58 examples, 0 failures.

## Post-deploy validation (from ticket AC)

| Window | Check | Action if tripped |
|---|---|---|
| Day 1-7 | Completion-rate funnel stays ≥ 70.82% (baseline 72.82% - 2pp tolerance) | Flip flag off, investigate |
| Day 1-30 | Fill rate ≥ 50% target; < 20% triggers review | < 20%: reconsider placement/wording in follow-up |
| Day 30-60 | Cross-reference selections against actual adoption timing in SF | Validates predictive value |

## Flag cleanup

Feature flag stays in place for at least 60 days post-launch (covers monitoring windows). Removal lands in a follow-up ticket.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

[DATA-301]: https://openstax.atlassian.net/browse/DATA-301?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/handlers/newflow/educator_signup/complete_profile.rb [hunk 1]
```diff
@@ -117,6 +124,13 @@ def handle
 
       private #################
 
+      def expected_start_semester
+        return nil if signup_params.expected_start_semester.blank?
+        return nil unless EXPECTED_START_SEMESTERS.include?(signup_params.expected_start_semester)
+        return nil unless [AS_PRIMARY, AS_RECOMMENDING].include?(signup_params.using_openstax_how)
+        signup_params.expected_start_semester
+      end
```
[the remark this answers is not in the document]
Comment (mwvolo) [reply app/handlers/newflow/educator_signup/complete_profile.rb:176]: Salesforce is already ready https://openstax.atlassian.net/browse/DATA-346

File: app/routines/newflow/create_or_update_salesforce_lead.rb [hunk 2]
```diff
@@ -150,6 +159,7 @@ def exec(user:)
       lead.subject_interest = user.which_books
       lead.num_students = user.how_many_students
       lead.adoption_status = ADOPTION_STATUS_FROM_USER[user.using_openstax_how]
+      lead.expected_start_semester = EXPECTED_START_SEMESTER_LABELS[user.expected_start_semester]
```
[the remark this answers is not in the document]
Comment (mwvolo) [reply app/routines/newflow/create_or_update_salesforce_lead.rb]: Salesforce is ready https://openstax.atlassian.net/browse/DATA-346

File: config/initializers/openstax_salesforce.rb [hunk 3]
```diff
@@ -2,6 +2,12 @@
 # also be copied to the application's initializers by running the install
 # task. Because this code can get run multiple times, make sure to only put
 # code here that is amenable to that.
+
+# Extend the gem's Lead class with fields not yet in the published gem version.
+OpenStax::Salesforce::Remote::Lead.field(
+  :expected_start_semester, from: 'Expected_Start_Semester__c'
+)
```
[the remark this answers is not in the document]
Comment (mwvolo) [reply config/initializers/openstax_salesforce.rb]: I should just update openstax_salesforce. I'll do that and remove this class extension

File: app/routines/newflow/create_or_update_salesforce_lead.rb [hunk 4]
```diff
@@ -14,6 +14,15 @@ class CreateOrUpdateSalesforceLead
 
     private_constant(:ADOPTION_STATUS_FROM_USER)
 
+    EXPECTED_START_SEMESTER_LABELS = {
+      this_semester:      'This semester',
+      next_semester:      'Next semester',
+      next_academic_year: 'Next academic year',
+      just_exploring:     'Just exploring'
+    }.with_indifferent_access.freeze
```
Comment (Dantemss) [inline app/routines/newflow/create_or_update_salesforce_lead.rb]: This has to be kept in sync with EXPECTED_START_SEMESTERS in another file which is not great. We could read the keys from this hash into that array. Or just add a comment to both places saying they have to be kept in sync.

File: config/initializers/openstax_salesforce.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (Dantemss) [reply config/initializers/openstax_salesforce.rb]: Updating the gem would be ideal. But the class probably won't reload, since it's in a gem.

File: config/locales/en.yml [hunk 5]
```diff
@@ -73,6 +73,12 @@ en:
     school_issued_email_must_be_entered: Please enter your school-issued email address
     school_issued_email_is_invalid: Please enter a valid email address
     school_issued_email_is_taken: Email already in use
+    expected_start_semester: When do you plan to start using OpenStax?
+    expected_start_semester_options:
+      this_semester: This semester
+      next_semester: Next semester
+      next_academic_year: Next academic year
+      just_exploring: Just exploring
```
Comment (Dantemss) [inline config/locales/en.yml:91]: Oh... maybe we could just use these instead of EXPECTED_START_SEMESTER_LABELS?

**Review by Dantemss — APPROVED** [review]
Looks good, see comments

File: config/initializers/openstax_salesforce.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (mwvolo) [reply config/initializers/openstax_salesforce.rb]: is there any reason not to just have this code live in the accounts code? 
with tutor gone, I doubt we'll have other rails apps using this and it might make it more maintainable.
